internal/fixture/state.go

```diff
@@ -182,8 +201,22 @@ var _ Operation = &Apply{}
 
 func (a Apply) run(state *State) error {
 	err := state.Apply(a.Object, a.APIVersion, a.Manager, false)
-	if (err != nil || a.Conflicts != nil) && !reflect.DeepEqual(err, a.Conflicts) {
-		return fmt.Errorf("expected conflicts: %v, got %v", a.Conflicts, err)
+	if err != nil && err.(merge.Conflicts) == nil {
+		return err
+	}
+	if a.Conflicts != nil {
+		conflicts := merge.Conflicts{}
+		if err != nil {
+			conflicts = err.(merge.Conflicts)
+		}
+		if len(addedConflicts(a.Conflicts, conflicts)) != 0 || len(addedConflicts(a.Conflicts, conflicts)) != 0 {
```

Comment:
x || x
both sides are the same

maybe:
```suggestion
		if len(addedConflicts(a.Conflicts, conflicts)) != 0 || len(addedConflicts(conflicts, a.Conflicts)) != 0 {
```